crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -1135,21 +1151,17 @@ where
                             },
                         );
 
-                        if &alias.node.name == "annotations" {
-                            self.ctx.annotations_future_enabled = true;
-                        }
```

Comment:
This got moved "up" into an earlier phase in `visit_stmt` -- the phase where we check if we're past the `__future__` and `import` boundaries.

The nice thing about this separation is that _only_ this first phase modifies any permanent flags. All subsequent modifications are local to the subtree, so we can save the current flags and restore them at the end of the visitor.